src/server/app.ts

```diff
@@ -69,18 +73,28 @@ export async function counterfact(config: Config) {
   const koaApp = createKoaApp(registry, middleware, config);
 
   // eslint-disable-next-line max-statements
-  async function start(options: { http?: boolean } = {}) {
-    const http = options.http ?? true;
-
-    await codeGenerator.watch();
-    await transpiler.watch();
-    await moduleLoader.load();
-    await moduleLoader.watch();
+  async function start(options: Config) {
+    const {
+      generate,
+      startRepl: shouldStartRepl,
+      startServer,
+      watch,
+    } = options;
+
+    if (generate.routes || generate.types) {
+      await codeGenerator.generate();
+    }
+    if (watch.routes || watch.types) {
+      await codeGenerator.watch();
+      await transpiler.watch();
```

Comment:
We need `transpiler` to run when the server is running (`--serve`). It's not generating code. It's transpiling TypeScript code that was edited by hand into JavaScript so Node can load it.

Reply:
Done!